### DISCUSSION TITLE
Disableable credentials feature not supported fully in default keycloak?

### DISCUSSION THREAD

**Discussion by MariuszCwikla** [keycloak/keycloak#12323] [Q&A]
Hi,
how can I use "Disableable credentials" feature? I have found this article https://ultimatesecurity.pro/post/2fa-disable/ that presents this.
Unfortunately I was not able to test this locally.
I am looking through the code and it seems that this is not implemented in the standard keycloak distro and is intended for custom storage providers (?). All existing implementations of `CredentialInputUpdaters` have blank implementation:

```
    @Override
    public void disableCredentialType(RealmModel realm, UserModel user, String credentialType) {

    }
```

I need to support following use case:
- user configures his TOTP, but this credential must not be "active" yet.
- there will be separate confirmation step that will activate TOTP credential later

I suppose that disableable credentials feature would help here (create credential and disable right away; enable later when I have the confirmation), but apparently I cannot use it out of the box.

Do you think that implementing custom `CredentialInputUpdater` is enough (or actually `UserStorageProvider`?).
Or would this be a good idea to add this feature to the product?

**Comment by workcomped** [comment]
From what I can gather, this may be a feature gap in the Admin v2 UI. The Admin REST API still has the ability to disable credentials type for a specific user but it is no longer in the React front-end. Perhaps the Keycloak team doesn't want to encourage this action? I'm not sure. I was given a requirement to support this behavior and having a UI option would be nice.
